app/src/main/java/rick_and_morty/data/repository/CharacterRepository.kt

```diff
@@ -3,14 +3,20 @@ package rick_and_morty.data.repository
 import rick_and_morty.di.RickAndMortyApiRemoteDataSource
 import javax.inject.Inject
 import rick_and_morty.data.model.CharacterResultsDto
+import rick_and_morty.data.event.BusEvent
 
 
 class CharacterRepository @Inject constructor(
     private val rickAndMortyApiRemoteDataSource: RickAndMortyApiRemoteDataSource
 ){
     
-   suspend fun getCharacters(page: Int): List<CharacterResultsDto> {
-       return rickAndMortyApiRemoteDataSource.fetchRickAndMortyData(page).results
+   suspend fun getCharacters(page: Int): BusEvent<List<CharacterResultsDto>> {
+       return try {
+           BusEvent.Success(list = rickAndMortyApiRemoteDataSource.fetchRickAndMortyData(page).results)
+       }
+       catch (error: Exception){
+           BusEvent.Error(exception = error)
+       }
     }
```

Comment:
I thought about this again and there are several drawbacks of this implementation:
1. The repository should just pass the data. We are doing similarly in the repositories of the `Welt News App`. We should handle errors in the view model and let the UI decide how to handle this and react on errors directly.
2. It is quite easy to lose overview over all the events that are created and handled. Especially when there will be a loading event added to the events being sent. This one will be most likely added in the view model. We should handle these kind of events in one place, preferably the view model and send the respective event/state to the UI (composables).

I would therefore suggest to just return `List<CharacterResultsDto>` here.

Reply:
Returned back as it was before.